diff --git a/field_collection_modals.module b/field_collection_modals.module
index f809e04..e43d28a 100644
--- a/field_collection_modals.module
+++ b/field_collection_modals.module
@@ -320,9 +320,10 @@ function field_collection_modals_operator_callback($node, $fieldname,
       $output = $commands;
     }
 
-    // Note:  using drupal_json_output here, but using this in the "add"
-    // function caused html 500 error
-    drupal_json_output($output);
+    // Render the output.
+    // Note:  using ajax_render here, using drupal_json_render here caused
+    // html 500 error
+    print ajax_render($output);
     drupal_exit();
   }
   else {
@@ -401,7 +402,7 @@ function field_collection_modals_field_attach_view_alter(&$output, $context) {
         if (is_numeric($key)) {
           // If Edit access, insert "edit" & "delete" links.
           if ($edit_access) {
-            if (count($output[$field]['#items'][$key]) == 1) {
+            if (count($output[$field]['#items'][$key]) != 0) {
 
               $output[$field][$key]['links']['#links']['edit']['href'] = 'fc/' .
               $context['entity']->nid . '/' . str_replace('_', '-', $field) . '/'
@@ -411,7 +412,7 @@ function field_collection_modals_field_attach_view_alter(&$output, $context) {
             $output[$field][$key]['links']['#links']['edit']['attributes']
               = array('class' => 'ctools-use-modal');
 
-            if (count($output[$field]['#items'][$key]) == 1) {
+            if (count($output[$field]['#items'][$key]) != 0) {
               $output[$field][$key]['links']['#links']['delete']['href'] = 'fc/' .
               $context['entity']->nid . '/' . str_replace('_', '-', $field) . '/'
               . $output[$field]['#items'][$key]['value'] . '/delete/nojs/go';
